File: nodejs/7 kyu/154.js
// https://www.codewars.com/kata/5a39724945ddce2223000800/train/javascript

// Passed

function totalAmountVisible(topNum, numOfSides) {
  let oppositeNum = numOfSides + 1 - topNum;
  let totalVisibleNums = 0;
  for (let currentNum = 1; currentNum <= numOfSides; currentNum++) {
    if (currentNum === oppositeNum) continue;
    totalVisibleNums += currentNum;
  }

  return totalVisibleNums;
}

const output = totalAmountVisible(3, 6);
console.log(output);

File: python/7 kyu/154.py
# https://www.codewars.com/kata/57d532d2164a67cded0001c7/train/python

# Passed

def histogram(results):
    dice_to_frequency_dict = {
        dice_num: frequency for dice_num, frequency in enumerate(results, start=1) 
    }
    
    histogram_string = ""
    for dice_num in range(6, 0, -1):
        frequency = dice_to_frequency_dict[dice_num]
        if frequency > 0:
            histogram_string += f"{dice_num}|{'#' * frequency} {frequency}".strip() + "\n"
        else:
            histogram_string += f"{dice_num}|{'#' * frequency}".strip() + "\n"
            
    return histogram_string

output = histogram([7,3,10,1,0,5])
print(output)